src/connector/src/source/filesystem/s3/source/reader.rs

```diff
@@ -65,9 +67,30 @@ impl S3FileReader {
         let split_id = split.id();
 
         let object_name = split.name.clone();
-        let byte_stream =
-            S3FileReader::get_object(&client_for_s3, &bucket_name, &object_name, split.offset)
-                .await?;
+
+        let byte_stream = match S3FileReader::get_object(
+            &client_for_s3,
+            &bucket_name,
+            &object_name,
+            split.offset,
+        )
+        .await
+        .map_err(|sdk_err| sdk_err.into_service_error())
+        {
+            Ok(s) => s,
+            Err(GetObjectError::NoSuchKey(_)) => {
+                tracing::warn!("S3 Object {} not found, ignoring", object_name);
+                return Ok(());
```

Comment:
should we also mark the file's state as done in state store to prevent looking for the file second time?

Reply:
Yes, the residual status still exists in the state store. The enumerator needs to initiate split reduction to push the delete command.

This is just a temporary solution, otherwise the deleted split will prevent the cluster from starting and it will remain in the recovery state.

Reply:
never mind. it is ok to leave it for now.